src/main/java/teammates/storage/api/FeedbackSessionsDb.java

```diff
@@ -194,14 +195,27 @@ public void addInstructorRespondents(List<String> emails, FeedbackSessionAttribu
             throw new InvalidParametersException(feedbackSession.getInvalidityInfo());
         }
 
-        FeedbackSession fs = getEntity(feedbackSession);
-        if (fs == null) {
-            throw new EntityDoesNotExistException(ERROR_UPDATE_NON_EXISTENT + feedbackSession.toString());
+        try {
+            ofy().transact(new VoidWork() {
+                @Override
+                public void vrun() {
+                    FeedbackSession fs = getEntity(feedbackSession);
+                    if (fs == null) {
+                        throw new RuntimeException(new EntityDoesNotExistException(
+                                ERROR_UPDATE_NON_EXISTENT + feedbackSession.toString()));
+                    }
+
+                    fs.getRespondingInstructorList().addAll(emails);
+
+                    saveEntity(fs, feedbackSession);
+                }
+            });
+        } catch (RuntimeException e) {
```

Comment:
This looks dangerous. Why catch a `RuntimeException` here?

Reply:
This is the only way to throw a checked exception from within the block of code to be run in a transaction: https://groups.google.com/d/msg/objectify-appengine/0NcX0KLNIDo/6ljC6M9-x98J

Reply:
Can't say I like this method, but we don't really have a say since this is a library restriction.